app.js

```diff
@@ -84,9 +81,12 @@ function createApp(config) {
   app.use(helmet())
   app.use(requestId())
   app.use(cachingMiddleware(caching()))
+  app.use('/schemas', express.static('./schemas'))
 
   app.use(morgan('dev'))
-  app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDoc))
+
+  const swaggerOptions = { swaggerUrl: 'http://localhost:4000/schemas/swagger.yaml' }
```

Comment:
@jeffmcaffer this is probably a minor, but is the hardcoded URL necessary?  The way I generally approach this is via the swagger-express-mw, and injecting a config like:
.```
.
const config = {
  appRoot: __dirname + '/server', // required config
  swaggerFile: __dirname + '/server/api/swagger/swagger.yaml'
};
...

swagger_mw.create(config, function (err, swagger_mw) {
  if (err) {
    console.error('Unable to create swagger-express-mw');
    throw err;
  }
```

But I understand we are using the standard express middlewares and if we don't have any use of the above, no need to import it.

Reply:
right, no this will not be hard coded, that is just where I got to in the experiments. Thanks for pointing it out. Note that the real problem here is that swagger-ui needs to be configured with a `swaggerUrl` if it is going to resolve remote $refs. So what we need here is the service's base URL. That is configured into the service via the `SERVICE_ENDPOINT` config property.